### PR TITLE
add read support for FOCMEC output files

### PR DESCRIPTION
**Pull request by megies**
### What does this PR do?

Add read support for focal mechanisms calculated using program FOCMEC (`out` and `lst` files)

### PR Checklist
- [x] Correct base branch selected? `master` for new features, `maintenance_...` for bug fixes
- [x] the PR is making changes to documentation   +DOCS
- [x] All tests still pass.
- [x] Any new features or fixed regressions are be covered via new tests.
- [x] Any new or changed features have are fully documented.
- [x] Significant changes have been added to `CHANGELOG.txt` .

---

```python
In [1]: cat = read_events('/path/to/focmec_8sta.lst')

In [2]: print(cat[0].focal_mechanisms[0])
FocalMechanism
	            resource_id: ResourceIdentifier(id="smi:local/ab869739-a73b-411f-b788-9d39e5edaf0a")
	           nodal_planes: NodalPlanes(nodal_plane_1=NodalPlane(strike=59.08, dip=76.43, rake=-64.23), nodal_plane_2=NodalPlane(strike=174.99, dip=28.9, rake=-150.97), preferred_plane=1)
	          azimuthal_gap: 236.7
	 station_polarity_count: 23
	                 misfit: 0.0
	          creation_info: CreationInfo(creation_time=UTCDateTime(2017, 9, 8, 14, 54, 58), version='FOCMEC')
	                   ---------
	               comments: 1 Elements

In [3]: print(cat[0].focal_mechanisms[0].comments[0].text)

     Dip,Strike,Rake     76.43    59.08   -64.23
     Dip,Strike,Rake     28.90   174.99  -150.97   Auxiliary Plane
     Lower Hem. Trend, Plunge of A,N     84.99    61.10   329.08    13.57
     Lower Hem. Trend, Plunge of P,T    358.82    51.71   128.90    26.95
     B trend, B plunge, Angle:  232.62  25.00 105.00

          Log10(Ratio)                              Ratio     S Polarity
     Observed  Calculated    Difference  Station     Type     Obs.  Calc. Flag
      0.8847      0.8950      -0.0103      BLA        SH       R      R       
      1.1785      1.0810       0.0975      COR        SH       R      R       
      0.6013      0.5442       0.0571      HRV        SH       R      R       
      0.3287      0.3666      -0.0379      KEV        SH       L      L       
      0.8291      0.9341      -0.1050      KIP        SH       R      R       
      0.8033      0.7815       0.0218      KIP        SV       B      B       
      1.0783      1.1857      -0.1074      PAS        SH       R      R       
      0.2576      0.2271       0.0305      TOL        SH       L      L       
     -0.2762     -0.4076       0.1314      TOL        SS       F      F    NUM
     -0.4283     -0.4503       0.0220      HRV        SS       F      F       
     -0.0830      0.0713      -0.1543      KEV        SS       B      B       

Total number of ratios used is  11
RMS error for the 11 acceptable solutions is 0.0852
  Highest absolute velue of diff for those solutions is  0.1543

```

### CODE REVIEW

**Comment by krischer** [comment]
Let us know once this is ready for review.

**Comment by megies** [comment]
Ready for review.. it's a bit ugly but this comes mostly from the `lst` files output by FOCMEC not being very well constrained. They can look pretty different depending on whether input is polarities only or ratios only or both, and also depending on the chosen settings when running FOCMEC. So for many things I have to look through the whole header / focal mechanism block with regex because we can not rely on positioning of information.

**Comment by megies** [comment]
Rebased

**Comment by megies** [comment]
Rebased on current master and force-pushed so that we have fresh CI results tomorrow.

File: obspy/io/focmec/core.py [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+import warnings
+
+import numpy as np
+
+from obspy import UTCDateTime, Catalog, __version__
+from obspy.core.event import (
+    Event, FocalMechanism, NodalPlanes, NodalPlane, Comment, CreationInfo)
+
+
+# XXX some current PR was doing similar, should be merged to
```
Comment (ThomasLecocq) [inline obspy/io/focmec/core.py]: what about this ? is this still valid?

File: obspy/io/focmec/core.py [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+# XXX core/utcdatetime.py eventually..
```
Comment (ThomasLecocq) [inline obspy/io/focmec/core.py]: idem

File: obspy/io/focmec/core.py (hunk 1 above)
Comment (megies) [reply obspy/io/focmec/core.py]: Yeah I think so, just a note for future reference I guess

**Comment by megies** [comment]
Test fails are unrelated.

**Pull request merged**